tests/test-mappings-filter.php

```diff
@@ -0,0 +1,213 @@
+		}
+	}
+
+	public function test_when_added_filters_should_correctly_produce_jsonld_output() {
+		$post_id = $this->factory()->post->create( array() );
+		wp_add_object_terms( $post_id, 'how-to', 'category' );
+		// create filters
+		$manual_properties = array(
+			array(
+				'property_name'      => '@type',
+				'field_type'         => 'text',
+				'field_name'         => 'HowTo',
+				'transform_function' => 'none',
+				'property_status'    => Mappings_Validator::ACTIVE_CATEGORY,
+				'mapping_id'         => 'foo'
+			),
+		);
+
+		$mapping_id = 'foo';
+
+		add_filter( 'wl_mappings_register_mappings', function ( $mappings ) use ( $mapping_id ) {
+			array_push( $mappings, array(
+				'mapping_id'     => $mapping_id,
+				'mapping_title'  => 'foo title',
+				'mapping_status' => 'active'
+			) );
+
+			return $mappings;
+		} );
+
+		add_filter( 'wl_mappings_register_rule_groups', function ( $rule_groups ) use ( $mapping_id ) {
+			array_push( $rule_groups, array(
+				'rule_group_id' => 'rg_recipe_1',
+				'mapping_id'    => $mapping_id
+			) );
+
+			return $rule_groups;
+		} );
+
+		add_filter( 'wl_mappings_register_rules', function ( $rules ) {
+			array_push( $rules, array(
+				'rule_field_one'   => 'category',
+				'rule_logic_field' => '===',
+				'rule_field_two'   => 'how-to',
+				'rule_group_id'    => 'rg_recipe_1',
+			) );
+
+			return $rules;
+		} );
+
+		add_filter( 'wl_mappings_register_properties', function ( $properties ) use ( $manual_properties ) {
+			$properties = array_merge( $properties, $manual_properties );
+
+			return $properties;
+		} );
+
+		// Get the json ld data for this post.
+		$jsonlds       = $this->jsonld_service->get_jsonld( false, $post_id );
+		$target_jsonld = end( $jsonlds );
+		$this->assertArrayHasKey( '@type', $target_jsonld );
+		$this->assertEquals( 'HowTo', $target_jsonld['@type'] );
+	}
+
+
+	public function test_registering_mapping_with_transformation_function_should_work() {
+		$post_id = $this->factory()->post->create( array() );
+		wp_add_object_terms( $post_id, 'how-to', 'category' );
+		// create filters
+		$transformation_function = new Test_Transform_function();
+		$manual_properties       = array(
+			array(
+				'property_name'      => '@type',
+				'field_type'         => 'text',
+				'field_name'         => 'HowTo',
+				'transform_function' => $transformation_function->get_name(),
+				'property_status'    => Mappings_Validator::ACTIVE_CATEGORY,
+				'mapping_id'         => 'foo'
+			),
+		);
+
+		$mapping_id = 'foo';
+
+		add_filter( 'wl_mappings_register_mappings', function ( $mappings ) use ( $mapping_id ) {
+			array_push( $mappings, array(
+				'mapping_id'     => $mapping_id,
+				'mapping_title'  => 'foo title',
+				'mapping_status' => 'active'
+			) );
+
+			return $mappings;
+		} );
+
+		add_filter( 'wl_mappings_register_rule_groups', function ( $rule_groups ) use ( $mapping_id ) {
+			array_push( $rule_groups, array(
+				'rule_group_id' => 'rg_recipe_1',
```

Comment:
@naveen17797 what happens if I provide a numeric id here? could it conflict with data from database?